Neos.ContentRepository.BehavioralTests/Tests/Parallel/ForkingDuringForking/ForkingDuringForkingTest.php

```diff
@@ -0,0 +1,213 @@
+            'Neos.ContentRepository:Root' => [],
+            'Neos.ContentRepository.Testing:Content' => [],
+            'Neos.ContentRepository.Testing:Document' => [
+                'properties' => [
+                    'title' => [
+                        'type' => 'string'
+                    ]
+                ],
+                'childNodes' => [
+                    'tethered-a' => [
+                        'type' => 'Neos.ContentRepository.Testing:Content'
+                    ],
+                    'tethered-b' => [
+                        'type' => 'Neos.ContentRepository.Testing:Content'
+                    ],
+                    'tethered-c' => [
+                        'type' => 'Neos.ContentRepository.Testing:Content'
+                    ],
+                    'tethered-d' => [
+                        'type' => 'Neos.ContentRepository.Testing:Content'
+                    ],
+                    'tethered-e' => [
+                        'type' => 'Neos.ContentRepository.Testing:Content'
+                    ]
+                ]
+            ]
+        ]);
+
+        $setupLockResource = fopen(self::SETUP_LOCK_PATH, 'w+');
+
+        $exclusiveNonBlockingLockResult = flock($setupLockResource, LOCK_EX | LOCK_NB);
+        if ($exclusiveNonBlockingLockResult === false) {
+            $this->log('waiting for setup');
+            if (!flock($setupLockResource, LOCK_SH)) {
+                throw new \RuntimeException('failed to acquire blocking shared lock');
+            }
+            $this->contentRepository = $this->contentRepositoryRegistry
+                ->get(ContentRepositoryId::fromString('test_parallel'));
+            $this->log('wait for setup finished');
+            return;
+        }
+
+        $this->log('setup started');
+        $contentRepository = $this->setUpContentRepository(ContentRepositoryId::fromString('test_parallel'));
+
+        $origin = OriginDimensionSpacePoint::createWithoutDimensions();
+        $contentRepository->handle(CreateRootWorkspace::create(
+            WorkspaceName::forLive(),
+            ContentStreamId::fromString('live-cs-id')
+        ));
+        $contentRepository->handle(CreateRootNodeAggregateWithNode::create(
+            WorkspaceName::forLive(),
+            NodeAggregateId::fromString('lady-eleonode-rootford'),
+            NodeTypeName::fromString(NodeTypeName::ROOT_NODE_TYPE_NAME)
+        ));
+        $contentRepository->handle(CreateNodeAggregateWithNode::create(
+            WorkspaceName::forLive(),
+            NodeAggregateId::fromString('nody-mc-nodeface'),
+            NodeTypeName::fromString('Neos.ContentRepository.Testing:Document'),
+            $origin,
+            NodeAggregateId::fromString('lady-eleonode-rootford'),
+            initialPropertyValues: PropertyValuesToWrite::fromArray([
+                'title' => 'title-original'
+            ])
+        ));
+
+        for ($i = 0; $i <= 500; $i++) {
```

Comment:
when i increase the number of nodes here, forking just gets really slow but the two processes never timeout.